src/main/webapp/components/workspaces/workspaces.js

```diff
@@ -59,6 +68,181 @@ const workspaceById = gql`
     }
   }
 `
+const queryAttributes = gql`
+  fragment QueryAttributes on MetacardAttributes {
+    title
+    cql
+    metacard_tags
+    metacard_type
+  }
+`
+
+const useCreateQuery = workspaceId => {
+  const [queryId, setQueryId] = React.useState('')
+  const mutation = gql`
+    mutation CreateQuery($attrs: MetacardAttributesInput!) {
+      createMetacard(attrs: $attrs) {
+        id: id
+        ...QueryAttributes
+      }
+    }
+    ${queryAttributes}
+  `
+
+  const saveMutation = gql`
+    mutation SaveWorkspace($id: ID!, $attributes: Json!) {
+      saveMetacardFromJson(id: $id, attributes: $attributes) {
+        ...WorkspaceAttributes
+      }
+    }
+    ${workspaceAttributes}
+  `
+
+  const [save] = useMutation(saveMutation)
+
+  const [create] = useMutation(mutation, {
+    onCompleted: data => {
+      save({
```

Comment:
We should be using the update option instead and writing to the apollo cache to avoid the need to add refetchQueries

Reply:
`refetchQueries` is a nice to thing have when you can use it. Normally in our system we can't because our data is stored in SOLR and with soft commit times, a refetchQueries would normally just return old data. However, when you query by id, solr is guaranteed to give you up to date data. In this specific case, the refetchQueries he's running is on a `workspaceById` query which means it will be fine.

I am of the opinion that if you don't have to write this goofy/complicated cache updating logic, you shouldn't have to.